apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -728,81 +752,83 @@ export class ApplicationController {
       })
     const templateId = existingApplication.typeId as ApplicationTypes
     const template = await getApplicationTemplateByTypeId(templateId)
-
     // TODO
     if (template === null) {
       throw new BadRequestException(
         `No application template exists for type: ${existingApplication.typeId}`,
       )
     }
+    return withCodeOwner(template.codeOwner, async () => {
+      const newAnswers = (updateApplicationStateDto.answers ?? {}) as FormValue
+      const namespaces = await getApplicationTranslationNamespaces(
+        existingApplication as BaseApplication,
+      )
+      const intl = await this.intlService.useIntl(namespaces, locale)
 
-    const newAnswers = (updateApplicationStateDto.answers ?? {}) as FormValue
-    const namespaces = await getApplicationTranslationNamespaces(
-      existingApplication as BaseApplication,
-    )
-    const intl = await this.intlService.useIntl(namespaces, locale)
+      const permittedAnswers =
+        await this.validationService.validateIncomingAnswers(
+          existingApplication as BaseApplication,
+          newAnswers,
+          user.nationalId,
+          false,
+          intl.formatMessage,
+        )
 
-    const permittedAnswers =
-      await this.validationService.validateIncomingAnswers(
+      await this.validationService.validateApplicationSchema(
         existingApplication as BaseApplication,
-        newAnswers,
-        user.nationalId,
-        false,
+        permittedAnswers,
         intl.formatMessage,
+        user,
       )
 
-    await this.validationService.validateApplicationSchema(
-      existingApplication as BaseApplication,
-      permittedAnswers,
-      intl.formatMessage,
-      user,
-    )
+      const mergedAnswers = mergeAnswers(
+        existingApplication.answers,
+        permittedAnswers,
+      )
 
-    const mergedAnswers = mergeAnswers(
-      existingApplication.answers,
-      permittedAnswers,
-    )
+      const mergedApplication: BaseApplication = {
+        ...(existingApplication.toJSON() as BaseApplication),
+        answers: mergedAnswers,
+      }
 
-    const mergedApplication: BaseApplication = {
-      ...(existingApplication.toJSON() as BaseApplication),
-      answers: mergedAnswers,
-    }
+      const {
+        hasChanged,
+        hasError,
+        error,
+        application: updatedApplication,
+      } = await this.applicationActionService.changeState(
```

Comment:
Would be ideal to wrap this with logging context, as it may invoke all kinds of things.